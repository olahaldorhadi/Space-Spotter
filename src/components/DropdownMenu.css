.dropdown {
    background-color: var(--offwhite);
    border-radius: 5px;
    margin: 5px;
    min-width: 150px;
    position: relative;
    display: inline-block;
}

.dropdown button {
    background-color: var(--offwhite);
    margin-top: 10px;
    margin-bottom: 10px;
    color: #111111;
    width: 100%;
    min-height: 40px;
}

.dropdown-menu {
    position: absolute;
    top: calc(100% + 3px);
    left: 0;
    width: 141px;
    background-color: var(--offwhite);
    color: #111111;
    width: 100%;
    border: 1px solid var(--orange);
    border-radius: 5px;
    z-index: 2;
}

.dropdown-menu button {
    display: block;
    margin: 0px;
    padding: 8px;
    border-radius: 0px;
    background-color: white;
}

.dropdown-menu button:hover {
    background-color: #cccccc;
}
