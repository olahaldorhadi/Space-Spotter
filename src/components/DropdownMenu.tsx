import { useState } from 'react';
import './DropdownMenu.css';

interface DropdownMenuProps {
    description: string;
    inputType: string;
    onChange: (value: string) => void;
}

export const campusList = [
    { key: 0, optionsText: "Gløshaugen" },
    { key: 1, optionsText: "Dragvoll" },
    { key: 2, optionsText: "Kalvskinnet" },
    { key: 3, optionsText: "Øya" },
    { key: 4, optionsText: "Tunga" },
    { key: 5, optionsText: "Moholt" },
    { key: 6, optionsText: "Olavskvartalet" },
    { key: 7, optionsText: "Solsiden" },
  ]

  export const typeList = [
    { key: 0, optionsText: "Grupperom" },
    { key: 1, optionsText: "Undervisningsrom" },
    { key: 2, optionsText: "Digital undervisning" },
    { key: 3, optionsText: "Auditorium" },
    { key: 4, optionsText: "Seminarrom" },
    { key: 5, optionsText: "Lesesal" },
    { key: 6, optionsText: "Vrimleareal" },
    { key: 7, optionsText: "Dataarbeidsplass" },
    { key: 8, optionsText: "Ferdighetsrom" },
    { key: 9, optionsText: "Boks" },
    { key: 10, optionsText: "Laboratorium" },
  ]

const DropdownMenu: React.FC<DropdownMenuProps> = ({ description, inputType, onChange }) => {
    const [isOpen, setIsOpen] = useState(false);
    const [currentDescription, setCurrentDescription] = useState(description);

    const toggleDropdown = () => {
        setIsOpen(!isOpen);
    };
    
    const handleButtonClick = (event: React.MouseEvent<HTMLButtonElement>) => {
        const updatedDescription = event.currentTarget.value || event.currentTarget.innerText;
        setCurrentDescription(updatedDescription);
        toggleDropdown();
        onChange(updatedDescription);
    }
    
    return (
        <>
            <div className='dropdown'>
                <button onClick={toggleDropdown}>{currentDescription}</button>
                {inputType === "campus" && isOpen && (
                    <div className='dropdown-menu'> 
                        {campusList.map(campus => (
                            <button onClick={handleButtonClick} key={campus.key}>{campus.optionsText}</button>
                        ))}
                    </div>
                )}
                {inputType === "roomtype" && isOpen && (
                    <div className='dropdown-menu'> 
                        {typeList.map(type => (
                            <button onClick={handleButtonClick} key={type.key}>{type.optionsText}</button>
                        ))}
                    </div>
                )}
            </div>
        </>
    )
}

export default DropdownMenu;

