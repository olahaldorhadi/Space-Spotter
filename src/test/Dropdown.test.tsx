import { describe, expect, it, vi } from 'vitest'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import DropdownMenu from '../components/DropdownMenu'

describe('DropdownMenu component', () => {
    it('renders DropdownMenu component with the initial description', () => {
        // Mock onChange function
        const mockOnChange = vi.fn()

        // Render the DropdownMenu component
        render(
            <DropdownMenu
                description="Initial Description"
                inputType="campus"
                onChange={mockOnChange}
            />
        )

        // Check if the button with the initial description is present
        const buttonElement = screen.getByText('Initial Description')
        expect(buttonElement).toBeInTheDocument()

        // Check if the dropdown menu is not initially open
        const dropdownMenu = screen.queryByTestId('dropdown-menu')
        expect(dropdownMenu).not.toBeInTheDocument()

        expect(document.body).toMatchSnapshot()
    })
})
