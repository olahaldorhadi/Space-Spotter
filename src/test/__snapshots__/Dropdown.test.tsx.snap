// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DropdownMenu component > renders DropdownMenu component with the initial description 1`] = `
<body>
  <div>
    <div
      class="dropdown"
    >
      <button>
        Initial Description
      </button>
    </div>
  </div>
</body>
`;
